src/index.ts

```diff
@@ -27,6 +27,16 @@ async function getSubPageURLs(filePath: string, selector: string) {
   return urls;
 }
 
+const downloadWithConcurrency = async (urls, concurrencyLimit = 5) => {
+  for (let i = 0; i < urls.length; i += concurrencyLimit) {
+    await Promise.all(
+      urls.slice(i, i + concurrencyLimit).map(async (url) => {
+        await downloadHTML(url, true);
+      })
+    );
+  }
```

Comment:
```suggestion
const downloadWithConcurrency = async (urls: string[], concurrencyLimit = 5) => {
  for (const list of splitArray(urls, concurrencyLimit))
    await Promise.all(
      list.map((url) => downloadHTML(url, true))
    ):
```
https://web-cell.dev/web-utility/functions/splitArray.html